Download.Mvc/Client/components/FileDownload.tsx

```diff
@@ -0,0 +1,35 @@
+﻿import * as React from 'react';
+import { match } from 'react-router-dom';
+import 'isomorphic-fetch';
+
+export class FileDownload extends React.Component<any, any>{
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            link: '',
+            loading: true
+        };
+    }
+    componentDidMount = () => {
+        // grab link to download
+        fetch('/api/downloadFile/' + this.props.id, { credentials: 'same-origin' })
+            .then(response => response.json())
+            .then(link => this.setState({ link }))
+            .then(loading => this.setState({ loading: false }))
```

Comment:
@laholstege you should do this in one line, `.then(link => this.setState({ link, loading: false }))`